CRM/Contribute/BAO/Contribution.php

```diff
@@ -5401,6 +5339,46 @@ protected static function getMultiplier($contribution_status_id, $context) {
   }
 
   /**
+   * Does this transaction reflect a payment instrument change.
+   *
+   * @param array $params
+   * @param array $pendingStatuses
+   *
+   * @return bool
+   */
+  protected static function isPaymentInstrumentChange(&$params, $pendingStatuses) {
+    $contributionStatus = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $params['contribution']->contribution_status_id);
+
+    if (array_key_exists('payment_instrument_id', $params)) {
+      if (CRM_Utils_System::isNull($params['prevContribution']->payment_instrument_id) &&
+        !CRM_Utils_System::isNull($params['payment_instrument_id'])
+      ) {
+        //check if status is changed from Pending to Completed
+        // do not update payment instrument changes for Pending to Completed
+        if (!($contributionStatus == 'Completed' &&
+          in_array($params['prevContribution']->contribution_status_id, $pendingStatuses))
+        ) {
+          return TRUE;
+        }
+      }
+      elseif ((!CRM_Utils_System::isNull($params['payment_instrument_id']) &&
+          !CRM_Utils_System::isNull($params['prevContribution']->payment_instrument_id)) &&
+        $params['contribution']->payment_instrument_id != $params['prevContribution']->payment_instrument_id
```

Comment:
checking non null value for payment_instrument_id from $params['payment_instrument_id'] and $params['prevContribution']->payment_instrument_id, but in next line comparing payment_instrument from 
$params['contribution']->payment_instrument_id

Reply:
@sunilpawar we are still comparing ```payment_instrument_id``` of 'contribution' and 'prevContribution' respectively. On which line we are comparing the ```payment_instrument_id``` with name ``````payment_instrument``` ?

Reply:
@monishdeb  can we connect Skype? i already send skype request.

Reply:
I think this was something well spotted by @sunilpawar & this should be merged 
https://github.com/civicrm/civicrm-core/pull/10980